src/components/App.js

```diff
@@ -111,16 +112,26 @@ export default class App extends React.Component {
     return event.key === ' ';
   }
 
-  canTogglePlayPause(event) {
+  canTogglePlayPause() {
     // Prevent play/pause toggle when elements with ids in the following list are pressed.
     const disallowedIds = [
       'recent-song-history',
       'toggle-play-pause',
       'stream-select',
-      'keyboard-controls'
+      'keyboard-controls',
+      'toggle-button-nav'
     ];
-    const [focusedElement] = event.path;
-    return !disallowedIds.includes(focusedElement.id);
+    return !disallowedIds.includes(document.activeElement.id);
+  }
+
+  isUpDownArrowPressed(event) {
+    return event.key === 'ArrowUp' || event.key === 'ArrowDown';
+  }
+
+  canAdjustVolume() {
+    // Ignore arrow hot keys if focus is on volume slider or stream selector.
+    const disallowedIds = ['volume-input', 'stream-select'];
```

Comment:
I am not sure why we would need to reject volume adjustment using shortcuts when the volume-input element is selected.

Other than that, LGMT. Thanks for the clean up.

Reply:
Good question. The volume slider input has an onChange handler which also adjusts the volume. So when the keyboard focus is on that input then you would get a double volume change, one for the onChange handler and one for the hotkey.